core/src/main/java/kafka/server/share/DelayedShareFetch.java

```diff
@@ -191,26 +191,23 @@ public void onExpiration() {
      * Complete the share fetch operation by fetching records for all partitions in the share fetch request irrespective
      * of whether they have any acquired records. This is called when the fetch operation is forced to complete either
      * because records can be acquired for some partitions or due to MaxWaitMs timeout.
+     * <p>
+     * On operation timeout, onComplete is invoked, last try occurs to acquire partitions and read
+     * from log, if acquired. The fetch will only happen from local log and not remote storage, on
+     * operation expiration.
```

Comment:
For better consistency, should we remove released partitions from `partitionsAcquired` in `releasePartitionLocks()`? It seems that if we had done that, we wouldn't get into the situation of releasing the lock more than once.

Reply:
I thought about it but it complicates the code execution in DelayedShareFetch. Clearing partitionsAcquired alone is not sufficient, as there could be pending remote fetch as well which might need additional time to finish. Hence I didn't go with that approach.